### PR TITLE
new post

### PR DESCRIPTION
**Pull request by recdevs**

### CODE REVIEW

File: _posts/2020-12-04-apex-batch-gotchas.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+      private Date executeDate;
+
+      public AccountActivationBatch(Date executeDate) {
+        this.executeDate = executeDate
+      }
+
+      // ... Batchable & Scheduleable implementation below
+
+The problem is on line 1: `implements Database.Batchable<sObject>, Schedulable`. Implementing both of these interfaces can lead to unexpected side effects. When an instance of a class is scheduled it gets saved & tucked away somewhere in Salesforce's backend. When execution time rolls around, Salesforce dusts off your class & calls the `execute` method. Do you see the problem in `AccountActivationBatch`?
```
Comment (chriskarpyszyn) [inline _posts/2020-12-04-apex-batch-gotchas.md]: tucked away in their backend ;)

File: _posts/2020-12-04-apex-batch-gotchas.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+Then it will always execute with the activation date of Nov 24th 2020. What is probably intended here is that the batch always executes with the current date.
+
+If a batch seems simple, perhaps because it is stateless, you may be tempted to 'save time' by making the batch schedulable. In this case you should be wary of how a less-wise developer might modify your batch to add state. It is best to program defensively and not allow your batch to be schedulable.
+
+### The Ungotcha
+
+Make your job easy and [separate concerns](https://en.wikipedia.org/wiki/Separation_of_concerns#:~:text=In%20computer%20science%2C%20separation%20of,code%20of%20a%20computer%20program.). Handle the batch and the scheduled execution of the batch separately. This way, you don't have to worry about stale property values. 
+
+Another benefit of this approach is that you are more free to upgrade the apex code of the batch. There are weird side-effects caused by updating a scheduled class. If the batch were scheduled then you would have to worry about those side effects.
```
Comment (chriskarpyszyn) [inline _posts/2020-12-04-apex-batch-gotchas.md]: What are the weird side-effects?

**Review by chriskarpyszyn — APPROVED** [review]

File: _posts/2020-12-04-apex-batch-gotchas.md (hunk 2 above)
Comment (recdevs) [reply _posts/2020-12-04-apex-batch-gotchas.md]: good point. I'll update

File: _posts/2020-12-04-apex-batch-gotchas.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
+---
+layout: post
+title:  "3 Apex Batch Gotchas"
+date:   2020-12-04
```
Comment (JonnyPower) [inline _posts/2020-12-04-apex-batch-gotchas.md]: let's update the date and get this out lol

**Review by JonnyPower — APPROVED** [review]

**Pull request merged**